spec/lib/whitehall_importer/integrity_checker/change_history_check_spec.rb

```diff
@@ -0,0 +1,114 @@
+RSpec.describe WhitehallImporter::IntegrityChecker::ChangeHistoryCheck do
+  let(:published_edition) { build(:edition, document_type_id: "news_story", state: "published") }
+  let(:draft_edition) { build(:edition, document_type_id: "news_story") }
+  let(:public_timestamp) { Time.zone.now.yesterday.rfc3339 }
+
+  describe "#match?" do
+    it "returns true if the proposed payload matches" do
+      proposed_change_history = [
+        {
+          note: "First published.",
+          public_timestamp: public_timestamp,
+        },
+      ]
+
+      publishing_api_change_history = [
+        {
+          note: "First published.",
+          public_timestamp: public_timestamp,
+        },
+      ]
+
+      integrity_check = described_class.new(published_edition, proposed_change_history, publishing_api_change_history)
+      expect(integrity_check.match?).to be true
+    end
+
+    it "returns true if the proposed payload matches for a draft edition with mismatched time stamps for first item" do
+      proposed_change_history = [
+        {
+          note: "First published.",
+          public_timestamp: public_timestamp,
+        },
+        {
+          note: "Updated",
+          public_timestamp: Time.zone.now.rfc3339,
+        },
+      ]
+
+      publishing_api_change_history = [
+        {
+          note: "First published.",
+          public_timestamp: Time.zone.now.rfc3339,
+        },
+        {
+          note: "Updated",
+          public_timestamp: Time.zone.now.rfc3339,
+        },
+      ]
+
+      integrity_check = described_class.new(draft_edition, proposed_change_history, publishing_api_change_history)
+      expect(integrity_check.match?).to be true
+    end
+
+    it "returns false if length of change history does not match" do
+      proposed_change_history = [
+        {
+          note: "First published.",
+          public_timestamp: public_timestamp,
+        },
+      ]
+
+      publishing_api_change_history = [
+        {
+          note: "First published.",
+          public_timestamp: public_timestamp,
+        },
+        {
+          note: "Updated",
+          public_timestamp: Time.zone.now.rfc3339,
+        },
+      ]
+
+      integrity_check = described_class.new(published_edition, proposed_change_history, publishing_api_change_history)
+      expect(integrity_check.match?).to be false
+    end
+
+    it "returns false if notes do not match" do
+      proposed_change_history = [
+        {
+          note: "First published.",
+          public_timestamp: public_timestamp,
+        },
+      ]
+
+      publishing_api_change_history = [
+        {
+          note: "Updated",
+          public_timestamp: public_timestamp,
+        },
+      ]
+
+      integrity_check = described_class.new(published_edition, proposed_change_history, publishing_api_change_history)
+      expect(integrity_check.match?).to be false
+    end
+
+    it "returns false if change history does not match" do
```

Comment:
I think it's the public_timestamp that we're testing rather than the whole change history item here?